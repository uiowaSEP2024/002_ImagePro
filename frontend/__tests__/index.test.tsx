// __tests__/index.test.jsx

import { render, screen, waitFor } from "@testing-library/react";
import Home from "@/pages/index";
import "@testing-library/jest-dom";
import { useRouter } from "next/router";

jest.mock('next/router', () => ({
  useRouter() {
    return ({
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn()
      },  beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null)
    });
  },
}));

const router = useRouter()

jest.mock('@/utils/auth', () => ({
  checkUserLoggedIn() {
    return new Promise((resolve) => {
      resolve( {detail : "already logged in!"} )
    });
  },
}));


// TODO: explore fixing snapshot testing with https://github.com/mui/material-ui/issues/21293#issuecomment-654921524
describe("Home", () => {
  it("renders a heading", async () => {
    render(<Home />);

    const heading = await waitFor(() =>
    screen.getByRole("heading", {
      name: /welcome to the tracking site/i,
    }));

    expect(heading).toBeInTheDocument();
  });
});
