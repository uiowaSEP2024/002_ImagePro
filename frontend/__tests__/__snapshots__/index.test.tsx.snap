// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <span
    aria-hidden="true"
    class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-igvLppd-css"
  />
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-ihLgHeV-css nextui-grid-item nextui-grid-container"
  >
    <h1
      align-items="center"
      class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
    >
      Welcome to the tracking site.
    </h1>
  </div>
</div>
`;
